### DISCUSSION TITLE
Is there a way to create a new empty Image (png)?

### DISCUSSION THREAD

**Discussion by foxnne** [nothings/stb#1453] [Q&A]
I'm using `stb` to read/write existing pngs without issue, but I'm writing an art program and need to generate a new image. Is this possible? I had thought allocating memory sized for `image_width * image_height * num_channels` and passing it along to `stbi_load_from_memory` would work but clearly the data is not correct.

Any information is appreciated, thank you!

**Comment by Manuzor** [comment]
> [...] I had thought allocating memory sized for image_width * image_height * num_channels [...]
Sounds like you want to preallocate and write to a buffer as you 
described and then use `stb_image_write.h` to save it to disk.
